search/tests/test_engines.py

```diff
@@ -0,0 +1,233 @@
+from .tests import MockSearchTests
+
+
+@override_settings(SEARCH_ENGINE="search.tests.utils.ForceRefreshElasticSearchEngine")
+class ElasticSearchTests(MockSearchTests):
+    """ Override that runs the same tests for ElasticSearchEngine instead of MockSearchEngine """
+
+    def test_reserved_characters(self):
+        """ Make sure that we handle when reserved characters were passed into query_string """
+        test_string = "What the ! is this?"
+        self.searcher.index("test_doc", {"content": {"name": test_string}})
+
+        response = self.searcher.search_string(test_string)
+        self.assertEqual(response["total"], 1)
+
+        response = self.searcher.search_string("something else !")
+        self.assertEqual(response["total"], 0)
+
+        response = self.searcher.search_string("something ! else")
+        self.assertEqual(response["total"], 0)
+
+        for char in RESERVED_CHARACTERS:
+            # previously these would throw exceptions
+            response = self.searcher.search_string(char)
+            self.assertEqual(response["total"], 0)
+
+    def test_facet_options(self):
+        """
+        Test that facet options work alongside facets  - notice unsupported in mock for now
+            size - is the only option for now
+        """
+        self._index_for_facets()
+
+        response = self.searcher.search()
+        self.assertEqual(response["total"], 7)
+        self.assertNotIn("facets", response)
+
+        facet_terms = {
+            "subject": {"size": 2},
+            "org": {"size": 2}
+        }
+        response = self.searcher.search(facet_terms=facet_terms)
+        self.assertEqual(response["total"], 7)
+        self.assertIn("facets", response)
+        facet_results = response["facets"]
+
+        self.assertEqual(facet_results["subject"]["total"], 6)
+        subject_term_counts = facet_results["subject"]["terms"]
+        self.assertEqual(subject_term_counts["mathematics"], 3)
+        self.assertEqual(subject_term_counts["physics"], 2)
+        self.assertNotIn("history", subject_term_counts)
+        self.assertEqual(facet_results["subject"]["other"], 1)
+
+        self.assertEqual(facet_results["org"]["total"], 7)
+        org_term_counts = facet_results["org"]["terms"]
+        self.assertEqual(org_term_counts["Harvard"], 4)
+        self.assertEqual(org_term_counts["MIT"], 2)
+        self.assertNotIn("edX", org_term_counts)
+        self.assertEqual(facet_results["org"]["other"], 1)
+
+
+@override_settings(MOCK_SEARCH_BACKING_FILE="./testfile.pkl")
+class FileBackedMockSearchTests(MockSearchTests):
+    """ Override that runs the same tests with file-backed MockSearchEngine """
+
+    def setUp(self):
+        MockSearchEngine.create_test_file()
+        self._searcher = None
+
+    def tearDown(self):
+        MockSearchEngine.destroy_test_file()
+        self._searcher = None
+
+    # Commenting test for now - it runs locally, but not on travis system
```

Comment:
Is this test being kept for local testing only?  If we're commenting it out basically for good, we should just remove it altogether.  If we want to keep it around, maybe add some more detail about the debugging scenario(s) in which someone might want to enable it.